privacy_evaluator/output/user_output.py

```diff
@@ -23,13 +22,53 @@ def __init__(
         """
         self.attack_data_y = attack_data_y
         self.privacy_risk = privacy_risk
-        self.all_labels = all_labels
 
     def histogram_top_k(
-        self, k: int = 10, show_diagram: bool = True
+        self, all_labels: np.ndarray, k: int = 10, show_diagram: bool = True
     ) -> Tuple[np.ndarray, np.ndarray]:
         """
         Draw histogram of class distribution of the k points with highest privacy risk score
+        :param all_labels: all the labels of the input data
+        :param k: the number of points to consider, default 10
+        :param show_diagram: determines if the diagram should be shown, default: True
+        :return: All lables of the data with the number of points that are in the top k
+        """
+        sorting = np.argsort(self.privacy_risk)
+        sorting = np.flip(sorting)
+        sorted_attack_data_labels = self.attack_data_y[sorting][:k]
+        df = pd.DataFrame(
+            {
+                "labels": sorted_attack_data_labels,
+            }
+        )
+        df = df.groupby("labels")["labels"].agg(["count"])
+        labels = df.index.to_numpy()
+        count = df["count"].to_numpy()
+        all_counts = np.array([])
+        for label in all_labels:
+            index = np.where(labels == label)
+            if len(index[0]) == 0:
+                all_counts = np.append(all_counts, 0)
+            else:
+                all_counts = np.append(all_counts, count[index[0]])
+        if show_diagram:
+            plt.bar(all_labels, all_counts)
+            plt.title("Histogram for top {} points per label".format(k))
+            plt.yticks(np.arange(0, np.int_(max(all_counts)) + 1, 1))
+            plt.show()
+        return all_labels, all_counts
+
+    def histogram_top_k_relative(
+        self,
+        label_counts: np.ndarray,
```

Comment:
Maybe calculate these values within the method instead of having a parameter here. But the rest looks fine :100: